### PR TITLE
Add aws-encryption-sdk Python Package

### PR DESCRIPTION
**Pull request by pathway27**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -12010,6 +12010,12 @@
     githubId = 248016;
     name = "Solène Rapenne";
   };
+  sree = {
+    email = "sreec@canva.com";
+    github = "pathway27";
+    githubId = 1563316;
+    name = "Sreekara Chelamalla";
+  };
```
Comment (pbsds) [inline maintainers/maintainer-list.nix:12018]: Please separate this change into a separate commit titled `maintainers: add sree`.

File: pkgs/development/python-modules/aws-encryption-sdk/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  ];
+
+  meta = {
+    description = "AWS Encryption SDK implementation for Python";
+    license = lib.licenses.asl20;
+    maintainers = with lib.maintainers; [ sree ];
+    homepage = "https://github.com/aws/aws-encryption-sdk-python";
+  };
+
+  doCheck = false;
```
Comment (pbsds) [inline pkgs/development/python-modules/aws-encryption-sdk/default.nix:38]: Disabling checks and including pytestCheckHook is pretty misleading. Please only do one or the other on the future.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 178541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.aws-encryption-sdk</li>
    <li>python39Packages.aws-encryption-sdk</li>
  </ul>
</details>

For fun i ran throught the broken tests for you:

```patch
diff --git a/pkgs/development/python-modules/aws-encryption-sdk/default.nix b/pkgs/development/python-modules/aws-encryption-sdk/default.nix
index 04130472871..c60319d2b2a 100644
--- a/pkgs/development/python-modules/aws-encryption-sdk/default.nix
+++ b/pkgs/development/python-modules/aws-encryption-sdk/default.nix
@@ -6,6 +6,8 @@
 , cryptography
 , attrs
 , boto3
+, pytest-mock
+, mock
 }:

 buildPythonPackage rec {
@@ -26,6 +28,20 @@ buildPythonPackage rec {

   checkInputs = [
     pytestCheckHook
+    pytest-mock
+    mock
+  ];
+  disabledTestPaths = [
+    # All of these (save one in the wildchar) requires a AWS token/credential
+    "examples/test/*.py"
+    "test/integration/test_i_thread_safety.py"
+    "test/integration/test_i_aws_encrytion_sdk_client.py"
+  ];
+  disabledTests = [
+    # touches network
+    "KMS key provider"
+    # requires AWS credentials
+    "Simple JS encrypt"
   ];

   meta = {
@@ -35,5 +51,4 @@ buildPythonPackage rec {
     homepage = "https://github.com/aws/aws-encryption-sdk-python";
   };

-  doCheck = false;
 }
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (pbsds) [inline maintainers/maintainer-list.nix:12018]: Sorry, forgot to mark it as "request changes"

**Review by pbsds — CHANGES_REQUESTED** [review]

**Comment by pathway27** [comment]
Bump

File: pkgs/development/python-modules/aws-encryption-sdk/default.nix [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+, boto3
+, pytestCheckHook
+, pytest-mock
+, mock
+}:
+
+buildPythonPackage rec {
+  pname = "aws-encryption-sdk";
+  version = "3.1.1";
+
```
Comment (pbsds) [inline pkgs/development/python-modules/aws-encryption-sdk/default.nix:16]:
```suggestion
  format = "setuptools";

```

File: pkgs/development/python-modules/aws-encryption-sdk/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  };
+
+  propagatedBuildInputs = [
+    wrapt
+    cryptography
+    attrs
+    boto3
+  ];
+
+  checkInputs = [
```
Comment (pbsds) [inline pkgs/development/python-modules/aws-encryption-sdk/default.nix:29]:
```suggestion
  nativeCheckInputs = [
```

**Review by pbsds — CHANGES_REQUESTED** [review]

**Comment by pbsds** [comment]
Great work. Could you please rebase the two commits such that the "maintainers: add sree" comes first?

**Comment by pbsds** [comment]
- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package build on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

---

Result of `nixpkgs-review pr 178541` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.aws-encryption-sdk</li>
    <li>python310Packages.aws-encryption-sdk.dist</li>
    <li>python311Packages.aws-encryption-sdk</li>
    <li>python311Packages.aws-encryption-sdk.dist</li>
  </ul>
</details>

**Review by pbsds — APPROVED** [review]

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by pathway27** [comment]
How to get this merged?

File: pkgs/development/python-modules/aws-encryption-sdk/default.nix [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "aws-encryption-sdk";
+  version = "3.1.1";
+  format = "setuptools";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0qwmgwzhakb8xdby97vafg3rcll0zlvx0kdymhf6p3f6iw0vypwd";
```
Comment (natsukium) [inline pkgs/development/python-modules/aws-encryption-sdk/default.nix:20]: Could you please use SRI hash?
> Nixpkgs contributors are currently recommended to use hash.

https://nixos.org/manual/nixpkgs/stable/#fetchurl

File: pkgs/development/python-modules/aws-encryption-sdk/default.nix [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    "test/integration/test_i_aws_encrytion_sdk_client.py"
+  ];
+
+  disabledTests = [
+    # touches network
+    "KMS key provider"
+    # requires AWS credentials
+    "Simple JS encrypt"
+   ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/aws-encryption-sdk/default.nix:49]: Could you please add `pythonImportsCheck`?
This check will be useful in the future, as it provides a simple check even if the check phase is skipped.

**Comment by wegank** [comment]
Done in #249286.
